File: test-openai.py
#!/usr/bin/env python3
"""
Simple OpenAI API test script for Python
This script tests if the OpenAI API key is working correctly
"""

import os
import sys
from openai import OpenAI

# Initialize OpenAI client with API key from environment variable
client = OpenAI(
    api_key=os.environ.get("OPENAI_API_KEY")
)

def test_openai():
    """Test if the OpenAI API key is working"""
    print("Testing OpenAI API connection...")
    
    try:
        # Simple API test - requesting a short message
        response = client.chat.completions.create(
            model="gpt-4o",  # the newest OpenAI model is "gpt-4o" which was released May 13, 2024
            messages=[
                {"role": "system", "content": "You are a helpful assistant."},
                {"role": "user", "content": "Hello from PanicSense! Can you confirm the OpenAI API is working?"}
            ],
            max_tokens=100
        )
        
        # Check if we got a valid response
        if response and response.choices and len(response.choices) > 0:
            print("✅ OpenAI API connection successful!")
            print(f"Response: {response.choices[0].message.content}")
            return True
        else:
            print("❌ OpenAI API returned an empty response")
            print(f"Response object: {response}")
            return False
    except Exception as e:
        print(f"❌ OpenAI API Error: {str(e)}")
        return False

if __name__ == "__main__":
    success = test_openai()
    if success:
        print("✅ API test completed successfully")
        sys.exit(0)
    else:
        print("❌ API test failed")
        sys.exit(1)

File: test-openai.js
/**
 * Simple OpenAI API test script
 * This script tests if the OpenAI API key is working correctly
 */

import { OpenAI } from 'openai';
import { fileURLToPath } from 'url';
import path from 'path';

// Initialize OpenAI client with API key from environment variable
const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY
});

async function testOpenAI() {
  console.log('Testing OpenAI API connection...');

  try {
    // Simple API test - requesting a short message
    const response = await openai.chat.completions.create({
      model: "gpt-4o", // the newest OpenAI model is "gpt-4o" which was released May 13, 2024
      messages: [
        { role: "system", content: "You are a helpful assistant." },
        { role: "user", content: "Hello from PanicSense! Can you confirm the OpenAI API is working?" }
      ],
      max_tokens: 100,
    });

    // Check if we got a valid response
    if (response && response.choices && response.choices.length > 0) {
      console.log('✅ OpenAI API connection successful!');
      console.log('Response:', response.choices[0].message.content);
      return true;
    } else {
      console.error('❌ OpenAI API returned an empty response');
      console.error('Response object:', response);
      return false;
    }
  } catch (error) {
    console.error('❌ OpenAI API Error:', error.message);
    if (error.response) {
      console.error('Error details:', error.response.data);
    }
    return false;
  }
}

// Execute the test if this is the main module
const isMainModule = process.argv[1] === fileURLToPath(import.meta.url);
if (isMainModule) {
  testOpenAI().then(success => {
    if (success) {
      console.log('✅ API test completed successfully');
    } else {
      console.error('❌ API test failed');
      process.exit(1);
    }
  }).catch(err => {
    console.error('Unexpected error:', err);
    process.exit(1);
  });
}

// Export for use in other files
export { testOpenAI };